### PR TITLE
Fix test results when the same request is executed more than 1 time (#2522)

### PR DESCRIPTION
**Pull request by BruAlcaraz**
# Description

This change resolves an issue when the runner execute the same request more than once. For example this happens when the request script sets the next request to itself. Before the changes the runner test results would look like this:
image

After this fix the result will show like this:
image

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

### CODE REVIEW

**Comment by BruAlcaraz** [comment]
@helloanoop  could you review this fix?

**Comment by BruAlcaraz** [comment]
@Its-treason and @dcoomber Could you guys review this fix?

**Comment by Its-treason** [comment]
If i'm understood your code correctly: You are trying to find the last Item instead of the first item? Could you use [`findLast`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast) instead? This would make the code a lot easier to read.

File: packages/bruno-app/src/providers/ReduxStore/slices/collections/index.js
Comment (martinsefcik) [inline packages/bruno-app/src/providers/ReduxStore/slices/collections/index.js]: Wouldn't be easier to use `.findLast((i) => i.uid === request.uid)` instead of cloning and reverting array x-times?

Comment (martinsefcik) [reply packages/bruno-app/src/providers/ReduxStore/slices/collections/index.js]: OK, now I see @Its-treason wrote the same few hours before me :)

Comment (BruAlcaraz) [reply packages/bruno-app/src/providers/ReduxStore/slices/collections/index.js]: Changed as suggested. No Idea why I over complicated on the first time....

**Comment by BruAlcaraz** [comment]
> If i'm understood your code correctly: You are trying to find the last Item instead of the first item? Could you use [`findLast`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findLast) instead? This would make the code a lot easier to read.

Changed as suggested. No Idea why I over complicated on the first time....

**Comment by BruAlcaraz** [comment]
@Its-treason can you please take a look again? I have done the suggested changes.

**Review by Its-treason — APPROVED** [review]

**Review by martinsefcik — APPROVED** [review]

**Comment by BruAlcaraz** [comment]
@helloanoop, @Its-treason and  @dcoomber what would be the next step to get these merged and released?

**Comment by Its-treason** [comment]
This is all you can do for now, you have to wait for the Maintainer to merge this. CC: @sanjai0py

**Pull request merged**
